src/pages/admin/events/[eventId]/index.tsx

```diff
@@ -0,0 +1,288 @@
+import CoreTypography from "src/components/core/typography";
+import VolQuickAddDialog from "src/components/VolQuickAddDialog";
+import InfiniteScroll from "react-infinite-scroll-component";
+import constants from "utils/constants";
+
+interface Props {
+    event: Event;
+    pageVols: PaginatedVolunteers;
+}
+
+function orderVolunteers(vols: Volunteer[]): Volunteer[] {
+    return vols
+        ? vols.sort((v1, v2) => {
+              // sorry typescript, need to sort
+              return Number(v1.name) - Number(v2.name);
+          })
+        : [];
+}
+
+const ManageVolunteers: NextPage<Props> = ({ pageVols, event }) => {
+    const styles = useStyles();
+    const [open, setOpen] = useState(false);
+    const [search, setSearch] = useState("");
+    const [vols, setVols] = useState<Volunteer[]>(pageVols.volunteers);
+    const [page, setPage] = useState<number>(1);
+    const [isLastPage, setIsLastPage] = useState<boolean>(false);
+
+    // helper func to get the vols by search query
+    async function getVolsFromSearch(query: string) {
+        const r = await fetch(`${urls.api.eventVolunteers(event._id, page)}&search=${query}`, {
+            method: "GET",
+        });
+        /* eslint-disable */
+        const response = await r.json();
+        const newVolsData: PaginatedVolunteers = response.payload;
+        /* eslint-enable */
+
+        setVols(newVolsData.volunteers);
+    }
+
+    // helper func to get the vols for a certain page
+    async function getVolsForPage(newPage: number) {
+        const r = await fetch(`${urls.api.eventVolunteers(event._id, newPage)}&search=${search}`, {
+            method: "GET",
+        });
+        if (Math.floor(r.status / 100) !== 2) {
+            alert(`ERROR: ${r.status}, ${r.statusText}`);
+        }
+        /* eslint-disable */
+        const response = await r.json();
+        const newVolsData: PaginatedVolunteers = response.payload;
+        /* eslint-enable */
+        if (newVolsData === undefined) {
+            setIsLastPage(true);
+        } else {
+            setVols(vols.concat(newVolsData.volunteers));
+        }
+        // setIsLastPage(newVolsData.isLastPage);
+    }
+
+    const handleSearchChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        setSearch(event.target.value);
+        setPage(1);
+        await getVolsFromSearch(event.target.value);
+    };
+    const handleLoadMore = async () => {
+        await getVolsForPage(page + 1);
+        setPage(page + 1);
+    };
+
+    const createAndRegisterVol = async function (vol: Volunteer) {
+        try {
+            const response = await fetch(`${urls.baseUrl}${urls.api.eventQuickadd(event._id!)}`, {
+                method: "POST",
+                body: JSON.stringify({
+                    ...vol,
+                }),
+            });
+            const responseJSON = (await response.json()) as { success: boolean; payload: unknown };
```

Comment:
You can use the `APIResponse` type here. I forget we have that pretty often

Reply:
:+1:  will commit that to dev